### ISSUE TITLE
No screening in gradient

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#1540]
Running a geometry optimization in direct mode.

SCF step reports
```
  ==> DirectJK: Integral-Direct J/K Matrices <==

    J tasked:                  Yes
    K tasked:                  Yes
    wK tasked:                  No
    Integrals threads:           8
    Schwarz Cutoff:          1E-12
```

whereas gradient step reports
```
  ==> DirectJKGrad: Integral-Direct SCF Gradients <==

    Gradient:                    1
    J tasked:                  Yes
    K tasked:                  Yes
    wK tasked:                  No
    Integrals threads:           8
    Schwarz Cutoff:          0E+00
```

The gradient step should use the same screening as the SCF.

This is with
```
          Psi4: An Open-Source Ab Initio Electronic Structure Package
                               Psi4 undefined 

                         Git: Rev {sadghost} cd5c9b3 dirty
```

**Comment by zachglick** [comment]
It looks like the default screening thresholds are different for energy and gradient calculations:

https://github.com/psi4/psi4/blob/3057799cde5524432ed3978416748386e437f1fb/psi4/src/psi4/libfock/jk.cc#L199

https://github.com/psi4/psi4/blob/3057799cde5524432ed3978416748386e437f1fb/psi4/src/psi4/scfgrad/jk_grad.cc#L118

I don't know if this is intentional, but it seems reasonable since a small integral doesn't guarantee a small gradient. Either way, these defaults are overridden when `ints_tolerance` is explicitly set.

**Comment by susilehtola** [comment]
> I don't know if this is intentional, but it seems reasonable since a small integral doesn't guarantee a small gradient.

Well, yes, but since the quartet is skipped over in the SCF, it shouldn't be included in the gradient either as the result is going to be different anyhow.

> Either way, these defaults are overridden when `ints_tolerance` is explicitly set.

That's just weird - you'd expect that default `ints_tolerance` to apply always.

**Comment by susilehtola** [comment]
Also, I don't think the gradients of the omitted quartets would be large.

Namely, if the gradient is large while the integral itself is tiny, then you'd have to have extremely localized basis functions that are relatively close together, just so that they have minimal overlap but their gradient is large.

This would basically require using very tight basis functions that float i.e. don't coincide with the nucleus.

Otherwise it's a non-physical issue. There's no reason *not* to screen the gradient integrals, especially since they're more costly than the usual integrals!

**Comment by susilehtola** [comment]
But yeah thanks @zachglick for pointing out the root cause. Indeed, if you specify `ints_tolerance` by hand, it works as it should.

Having the default gradient threshold be 0 is just confusing, since it gets changed when you manually specify the tolerance.

**Comment by dgasmith** [comment]
Closed in #1542.
